packages/office-ui-fabric-react/src/components/Layer/Layer.base.tsx

```diff
@@ -0,0 +1,177 @@
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+/* tslint:enable:no-unused-variable */
+
+import { Fabric } from '../../Fabric';
+import {
+  ILayerProps,
+  ILayerStyleProps,
+  ILayerStyles,
+} from './Layer.types';
+import {
+  css,
+  BaseComponent,
+  classNamesFunction,
+  customizable,
+  getDocument,
+  setVirtualParent
+} from '../../Utilities';
+
+let _layersByHostId: { [hostId: string]: LayerBase[] } = {};
+let _defaultHostSelector: string | undefined;
+
+const getClassNames = classNamesFunction<ILayerStyleProps, ILayerStyles>();
+
+// @customizable('Layer', ['theme'])
+export class LayerBase extends BaseComponent<ILayerProps, {}> {
+
+  public static defaultProps: ILayerProps = {
+    onLayerDidMount: () => undefined,
+    onLayerWillUnmount: () => undefined
+  };
+
+  private _rootElement: HTMLElement;
+  private _host: Node;
+  private _layerElement: HTMLElement | undefined;
+  private _hasMounted: boolean;
+  /**
+   * Used for notifying applicable Layers that a host is available/unavailable and to re-evaluate Layers that
+   * care about the specific host.
+   */
+  public static notifyHostChanged(id: string) {
+    if (_layersByHostId[id]) {
+      _layersByHostId[id].forEach(layer => layer.forceUpdate());
+    }
+  }
+
+  /**
+   * Sets the default target selector to use when determining the host in which
+   * Layered content will be injected into. If not provided, an element will be
+   * created at the end of the document body.
+   *
+   * Passing in a falsey value will clear the default target and reset back to
+   * using a created element at the end of document body.
+   */
+  public static setDefaultTarget(selector?: string) {
+    _defaultHostSelector = selector;
+  }
+
+  constructor(props: ILayerProps) {
+    super(props);
+
+    this._warnDeprecations({
+      onLayerMounted: 'onLayerDidMount'
+    });
+
+    if (this.props.hostId) {
+      if (!_layersByHostId[this.props.hostId]) {
+        _layersByHostId[this.props.hostId] = [];
+      }
+
+      _layersByHostId[this.props.hostId].push(this);
+    }
+  }
+
+  public componentDidMount() {
+    this.componentDidUpdate();
+  }
+
+  public componentWillUnmount() {
+    this._removeLayerElement();
+
+    if (this.props.hostId) {
+      _layersByHostId[this.props.hostId] = _layersByHostId[this.props.hostId].filter(layer => layer !== this);
+      if (!_layersByHostId[this.props.hostId].length) {
+        delete _layersByHostId[this.props.hostId];
+      }
+    }
+  }
+
+  public componentDidUpdate() {
+    let host = this._getHost();
+
+    const { getStyles, theme } = this.props;
+    const classNames = getClassNames(getStyles!,
+      {
+        theme: theme!,
```

Comment:
`this.props.className`  is never used, so that won't ever get added, even though it's in `props` and `layerStyleProps`